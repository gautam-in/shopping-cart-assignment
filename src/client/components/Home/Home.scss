.home-content {
    width: $default-width;
    margin: 0 auto;
    @media only screen and (max-width: 600px) {
        width: 100% ;
    }
}