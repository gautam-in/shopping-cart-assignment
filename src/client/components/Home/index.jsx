import React, { useEffect } from 'react';
import { map } from 'lodash';
import { useSelector, useDispatch } from 'react-redux';
import withStyles from 'isomorphic-style-loader/withStyles';

import styles from './Home.scss';

import Banner from '../Shared/Banner';
import ContentSection from '../Shared/ContentSection';
import bannerAction from '../../redux/actions/banner';
import categoryAction from '../../redux/actions/category';

const Home = () => {
  const dispatch = useDispatch();
  const banners = useSelector((state) => state.banners);
  const categories = useSelector((state) => state.categories);

  useEffect(() => {
    dispatch(bannerAction.bannerList());
    dispatch(categoryAction.categoryList());
  }, [dispatch]);

  return (
    <div className="home-content">
      <Banner banners={banners} />
      {
        map(categories, (category, cidx) => (
          category.order > -1
          && (
            <ContentSection
              key={cidx}
              category={category}
            />
          )
        ))
      }

    </div>
  );
};

export default {
  component: withStyles(styles)(Home),
  loadData: ({ dispatch }) => (async () => {
    await dispatch(bannerAction.bannerList());
    await dispatch(categoryAction.categoryList());
  })(),
};
