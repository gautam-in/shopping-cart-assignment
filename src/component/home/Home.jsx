import React, { Suspense } from "react";
import "./home.scss";
const CarouselHome = React.lazy(() => import("./CarouselHome"));
const Categories = React.lazy(() => import("./Categories"));

function Home() {
  return (
    <main className="app-home">
      <h1 style={{ display: "none" }}>Sabka Bazar shopping App</h1>
      <Suspense
        fallback={
          <div className="loader-block">
            <div className="loader"></div>
          </div>
        }
      >
        <a className="skip-line" href="#main">
          Skip to content
        </a>
        <CarouselHome />
        <Categories />
      </Suspense>
    </main>
  );
}

export default Home;
