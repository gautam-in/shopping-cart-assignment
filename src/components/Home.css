.body-section {
  width: 75vw;
  margin: 10px auto;
}
@media (min-width: 768px) and (max-width: 1024px) {
  .body-section {
    width: auto;
  }
}

@media (min-width: 768px) and (max-width: 1024px) and (orientation: landscape) {
  .body-section {
    width: auto;
  }
}

@media (min-width: 481px) and (max-width: 767px) {
  .body-section {
    width: auto;
  }
}
@media (min-width: 320px) and (max-width: 480px) {
  .body-section {
    width: auto;
  }
}
