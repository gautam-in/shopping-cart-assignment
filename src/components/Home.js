import React, { Component } from "react";
import axios from "axios";
import HomeBanners from "./HomeBanners";
import { isEmpty } from "lodash";

class Home extends Component {
  constructor(props) {
    super(props);

    this.state = {
      bannerList: [],
      categiryList: [],
      noDataText: "",
    };
  }

  componentDidMount() {
    this.bannerListData();
    this.categiriesListData();
  }

  bannerListData = () => {
    axios.get(`http://localhost:5000/banners`).then((res) => {
      if (res && res.status === 200) {
        const response = res.data;
        this.setState({
          bannerList: response,
        });
      } else {
        this.setState({
          bannerList: [],
        }).catch((err) => {
          this.setState({
            noDataText: "Something went wrong Please Try Again Later!",
          });
        });
      }
    });
  };

  categiriesListData = () => {
    axios
      .get(`http://localhost:5000/categories`)
      .then((res) => {
        if (res && res.status === 200) {
          const response = res.data;
          this.setState({ categiryList: response });
        } else {
          this.setState({
            categiryList: [],
            noDataText: "Something went wrong Please Try Again Later!",
          });
        }
      })
      .catch((err) => {
        this.setState({
          noDataText: "Something went wrong Please Try Again Later!",
        });
      });
  };

  handleExplore = (categoryId) => {
    this.props.history.push(`/products#${categoryId}`);
  };

  render() {
    return (
      <main>
        {!isEmpty(this.state.bannerList) ? (
          <HomeBanners bannerList={this.state.bannerList} />
        ) : (
          <p></p>
        )}
        {!isEmpty(this.state.categiryList) ? (
          <ul className="category-list">
            {this.state.categiryList.map((category) => (
              <li key={category.id} className="category">
                <article className="category-details">
                  <div className="category-title">{category.name}</div>
                  <div className="category-description">
                    {category.description}
                  </div>
                  <button
                    type="button"
                    className="category-explore-button"
                    onClick={() => this.handleExplore(category.id)}
                    tabIndex={0}
                    disabled={!category.enabled}
                    onKeyPress={() => this.handleExplore(category.id)}
                  >
                    Explore {category.name}
                  </button>
                </article>
                <article className="category-right">
                  <img
                    className="category-image"
                    src={category.imageUrl}
                    alt={category.name}
                    height="150"
                    width="200"
                  />
                </article>
              </li>
            ))}
          </ul>
        ) : (
          <h2
            style={{
              textAlign: "center",
              height: "70vh",
              display: "flex",
              justifyContent: "center",
              alignItems: "center",
            }}
          >
            {this.state.noDataText
              ? this.state.noDataText
              : "Data Not Avaliable"}
          </h2>
        )}
      </main>
    );
  }
}

export default Home;
