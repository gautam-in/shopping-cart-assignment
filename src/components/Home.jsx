import React, { Component } from 'react';
import "../styles/index.scss"

import CategoriesSection from "./CategoriesSection"
import SlickSlider from "./SlickSlider"


class Home extends Component {
  render() {
    return (
        <>
          <SlickSlider />
          <CategoriesSection />
        </>
    );
  }
}

export default Home;