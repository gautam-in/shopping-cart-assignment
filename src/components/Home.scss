@import '../SCSS-Variables/variables.scss';

.body-section {
  width: 75vw;
  margin: 10px auto;
}
@media (min-width: $min-width1) and (max-width: $max-width1) {
  .body-section {
    width: auto;
  }
}

@media (min-width: $min-width1) and (max-width: $max-width1) and (orientation: landscape) {
  .body-section {
    width: auto;
  }
}

@media (min-width: $width4) and (max-width: $width3) {
  .body-section {
    width: auto;
  }
}
@media (min-width: $min-width) and (max-width: $max-width) {
  .body-section {
    width: auto;
  }
}
