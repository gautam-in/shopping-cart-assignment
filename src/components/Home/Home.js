import OfferSlider from "../slider/OfferSlider";
import MainWrap from "../wrapper/Wrapper";
import Categories from "../categories/Categories";

const Home = () => {
  return (
    <MainWrap>
      <OfferSlider />
      <Categories />
    </MainWrap>
  );
};

export default Home;
