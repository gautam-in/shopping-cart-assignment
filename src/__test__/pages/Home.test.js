import { shallow,configure } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import Home from "../../components/pages/Home";
import SliderComponent from '../../components/SliderComponent';

configure({ adapter: new Adapter() })


describe('Home render', () => {
    it('should render our Snapshots correctly', () => {
        const wrapper = shallow(<Home />);
        expect(wrapper).toMatchSnapshot();
    });

    
    it('should render SliderComponent', () => {
        const wrapper = shallow(<Home />);
        const sliderComponent = wrapper.find(SliderComponent);
    expect(sliderComponent.exists()).toBe(true);
    })
});