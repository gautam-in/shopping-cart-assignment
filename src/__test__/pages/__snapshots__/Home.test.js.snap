// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator Snapshots should render our Snapshots correctly 1`] = `ShallowWrapper {}`;

exports[`Home render should render our Snapshots correctly 1`] = `ShallowWrapper {}`;
